view/view.go

```diff
@@ -154,107 +127,92 @@ func LiveView(id string, data *model.GetMeasurement, ctx model.Context, m model.
-	return ctx.Cmd == "http" && m.Options != nil && m.Options.Request != nil && m.Options.Request.Method == "GET" && !ctx.Full
-}
+	// Join lines back into a string
+	txt := strings.Join(lines, "\n")
 
-func OutputResults(id string, ctx model.Context, m model.PostMeasurement) {
-	fetcher := client.NewMeasurementsFetcher(client.ApiUrl)
+	return txt
+}
 
-	// Wait for first result to arrive from a probe before starting display (can be in-progress)
-	data, err := fetcher.GetMeasurement(id)
-	if err != nil {
-		fmt.Println(err)
-		return
-	}
+// Generate header that also checks if the probe has a state in it in the form %s, %s, (%s), %s, ASN:%d
+func generateHeader(result *model.MeasurementResponse, useStyling bool) string {
+	var output strings.Builder
 
-	// Probe may not have started yet
-	for len(data.Results) == 0 {
-		time.Sleep(apiPollInterval)
-		data, err = fetcher.GetMeasurement(id)
-		if err != nil {
-			fmt.Println(err)
-			return
-		}
-	}
+	// Continent + Country + (State) + City + ASN + Network + (Region Tag)
+	output.WriteString(getLocationText(result))
 
-	if ctx.CI || ctx.JsonOutput || ctx.Latency {
-		// Poll API until the measurement is complete
-		for data.Status == "in-progress" {
-			time.Sleep(apiPollInterval)
-			data, err = fetcher.GetMeasurement(id)
-			if err != nil {
-				fmt.Println(err)
-				return
+	// Check tags to see if there's a region code
+	if len(result.Probe.Tags) > 0 {
+		for _, tag := range result.Probe.Tags {
+			// If tag ends in a number, it's likely a region code and should be displayed
+			if _, err := strconv.Atoi(tag[len(tag)-1:]); err == nil {
+				output.WriteString(" (" + tag + ")")
+				break
 			}
 		}
+	}
 
-		if ctx.Latency {
-			OutputLatency(id, data, ctx)
-			return
-		}
-
-		if ctx.JsonOutput {
-			OutputJson(id, fetcher, ctx)
-			return
-		}
-
-		if ctx.CI {
-			PrintStandardResults(id, data, ctx, m)
-			return
-		}
+	headerWithFormat := formatWithLeadingArrow(output.String(), useStyling)
+	return headerWithFormat
+}
 
-		panic(fmt.Sprintf("case not handled. %+v", ctx))
+func formatWithLeadingArrow(text string, useStyling bool) string {
+	if useStyling {
+		return terminalLayoutArrow + terminalLayoutHighlight.Render(text)
 	}
+	return "> " + text
+}
 
-	LiveView(id, data, ctx, m)
+func isBodyOnlyHttpGet(ctx model.Context, m model.PostMeasurement) bool {
+	return ctx.Cmd == "http" && m.Options != nil && m.Options.Request != nil && m.Options.Request.Method == "GET" && !ctx.Full
 }
 
 func shareMessage(id string) string {
 	return fmt.Sprintf("View the results online: https://www.jsdelivr.com/globalping?measurement=%s", id)
 }
+
+func getLocationText(m *model.MeasurementResponse) string {
+	state := ""
+	if m.Probe.State != "" {
+		state = "(" + m.Probe.State + "), "
+	}
+	return m.Probe.Continent +
+		", " + m.Probe.Country +
+		", " + state + m.Probe.City +
+		", ASN:" + fmt.Sprint(m.Probe.ASN) +
+		", " + m.Probe.Network
```

Comment:
The network names can be very long and often break a single row into multiple lines. @jimaek we need to decide how to handle this:

A. Detect terminal width, compute the max width, and truncate the text with`...` at the end.
B. Remove network names from table output, show only physical location + ASN and hope that's short enough.

Reply:
Option A is the best. I've them being cut in other tools as well. Showing the first part can often be very useful even if the last part is cut

Reply:
I agree, it's going to be considerably harder probably, but much nicer for sure.